internal/resolver/dns/dns_resolver_test.go

```diff
@@ -1290,3 +1290,45 @@ func (s) TestMinResolutionInterval(t *testing.T) {
 		r.ResolveNow(resolver.ResolveNowOptions{})
 	}
 }
+
+// TestMinResolutionInterval_NoExtraDelay verifies that there is no extra delay
+// between two resolution requests apart from [MinResolutionInterval]
+// i.e. if a request is made after [MinResolutionInterval], it should resolve
+// immediately. Test sets [MinResolutionInterval] to 1 second and calls
+// ResolveNow() 4 times with a wait of 1 second between each request.
+// A correct implementation should resolve all requests without timing out in
+// test duration of 5 seconds.
```

Comment:
nit: The implementation details of the test can be mentioned in comments inside the test, only an overview is sufficient for the doc string.

Reply:
The following lines mention "what" the function is doing and not "why" its doing something.
>Test sets [MinResolutionInterval] to 1 second and calls ResolveNow() 4 times with a wait of 1 second between each request. A correct implementation should resolve all requests without timing out in test duration of 5 seconds.

Based on the [style guide]:
>It is also important to be aware that some attempts to provide clarity (such as adding extra commentary) can actually obscure the code’s purpose by adding clutter, restating what the code already says, contradicting the code, or adding maintenance burden to keep the comments up-to-date. Allow the code to speak for itself (e.g., by making the symbol names themselves self-describing) rather than adding redundant comments. It is often better for comments to explain why something is done, not what the code is doing.

Reply:
When I rewrote a whole bunch of tests, I eventually landed on the following approach:
- the test level comment would say at a very high level *what* the test is testing. It will also say at a very high level *what* operations the test is performing, and *what* is expects to see.
  - this helps readers of the test to get to know what is going on very quickly (and without reading the test) 
- then, i either put in comments or calls to `t.Log` for every logical block in the test
  - eg: start a management server, start test backends, configure `pick_first` LB policy on the channel, make RPC and expect it to fail with UNAVAILABLE etc
  - this approach also allows the reader to follow the test along very quickly without reading the code
  - and using `t.Logf` instead of comments sometimes can make debugging easy, since you will see those high level events in the test output when it fails

Reply:
> the test level comment would say at a very high level what the test is testing. It will also say at a very high level what operations the test is performing, and what is expects to see.

I think the existing comments more a less follows this